packages/admin/admin/src/hooks/useAsyncOptionsProps.ts

```diff
@@ -12,28 +12,19 @@ export function useAsyncOptionsProps<T>(loadOptions: () => Promise<T[]>): AsyncO
     const [open, setOpen] = React.useState(false);
     const [options, setOptions] = React.useState<T[]>([]);
     const loading = open && options.length === 0;
-    React.useEffect(() => {
-        let active = true;
-        if (!open) {
-            return undefined;
-        }
-        setOptions([]);
-        (async () => {
-            const response = await loadOptions();
-            if (active) {
-                setOptions(response);
-            }
-        })();
-        return () => {
-            active = false;
-        };
-    }, [loadOptions, open]);
+
+    const handleOpen = async () => {
+        setOpen(true);
+        setOptions([]); // Reset options to show loading
+        setOptions(await loadOptions());
+    };
```

Comment:
Fetching in the event handler is the better solution IMO. Similar to https://react.dev/learn/you-might-not-need-an-effect#sending-a-post-request